src/app/shared/utils/utils.ts

```
@@ -352,35 +352,47 @@ export class Util {
-
-    if (keys1.length !== keys2.length) {
+    if ((this.isEmpty(obj1) && !this.isEmpty(obj2)) || (this.isEmpty(obj2) && !this.isEmpty(obj1))) {
       return false;
     }
 
-    for (const key of keys1) {
-      const val1 = obj1[key];
-      const val2 = obj2[key];
-      if (typeof val1 === 'object' && val1 !== null && typeof val2 === 'object' && val2 !== null) {
-        if (!Util.deepEqual(val1, val2)) {
+    if (Array.isArray(obj1) && Array.isArray(obj2)) {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)